src/language-markdown/embed.js

```diff
@@ -57,7 +57,35 @@ function embed(path, options) {
     // MDX
     case "import":
     case "export":
-      return (textToDoc) => textToDoc(node.value, { parser: "babel" });
+      return async function (textToDoc) {
+        let isValidImportExport = false;
+
+        const doc = await textToDoc(node.value, {
+          // TODO: Rename this option since it's not used in HTML
+          __onHtmlBindingRoot(node) {
+            isValidImportExport = (() => {
+              const program = node.type === "File" ? node.program : node;
+              const { body } = program;
+              if (body.length !== 1) {
+                return false;
+              }
+
+              const [declaration] = body;
+              return (
+                (node.type === "import" &&
+                  declaration.type === "ImportDeclaration") ||
+                (node.type === "export" &&
+                  (declaration.type === "ExportDefaultDeclaration" ||
+                    declaration.type === "ExportNamedDeclaration" ||
+                    declaration.type === "ExportAllDeclaration"))
+              );
+            })();
```

Comment:
Added this check so we won't format unexpected unexpected code like that is valid JS code, but not a import declaration.

```
import .meta.url;
```